cmake/detray-functions.cmake

```diff
@@ -19,13 +19,19 @@ function( detray_add_library fullname basename )
    # Set up how clients should find its headers.
    target_include_directories( ${fullname} INTERFACE
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}> )
+      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}> )
 
    # Make sure that the library is available as "detray::${basename}" in every
    # situation.
    set_target_properties( ${fullname} PROPERTIES EXPORT_NAME ${basename} )
    add_library( detray::${basename} ALIAS ${fullname} )
 
+   # Specify the (SO)VERSION of the library.
+   set_target_properties( ${fullname} PROPERTIES
+      VERSION ${PROJECT_VERSION}
+      SOVERSION ${PROJECT_VERSION_MAJOR} )
```

Comment:
This is not harmful, but also not necessary.

The `VERSION` and `SOVERSION` properties really only make a difference for shared libraries. For all other library types (including `INTERFACE` libraries) these make no difference.

Still, as long as you added it, let's keep it...